### PR TITLE
Fix duplicate cards

### LINKED ISSUE #45: [BUG] Prevent Duplicate Cards in Game
## Description

The same card can appear as both reference and current card in sequence. The `/api/cards/next` endpoint uses random selection without exclusion.

## Steps

1. Open `packages/express-backend/api/cards/next.js`
2. Add query parameter support for `excludeId`
3. Modify aggregation pipeline to exclude the provided card ID
4. Open `packages/react-frontend/src/pages/GamePage.jsx`
5. Pass current reference card ID when fetching next card

## Files to Modify

- `packages/express-backend/api/cards/next.js`
- `packages/react-frontend/src/pages/GamePage.jsx`

## Expected Outcome

Next card fetched is never the same as the current reference card.

## Acceptance Criteria

- API accepts `excludeId` parameter
- Frontend passes reference card ID when fetching next card
- No duplicate cards appear in sequence

### PR DESCRIPTION
**Pull request by kevinrutledge**
## Developer: Kevin Rutledge

Closes #45

### Pull Request Summary

Implement frontend deck management to eliminate duplicate cards during gameplay sessions. Previously, the `/api/cards/next` endpoint used random selection that allowed the same cards to appear multiple times, creating poor user experience. This change fetches all cards once at game start, shuffles them locally, and serves cards sequentially from the deck so that no duplicates appear until the deck is exhausted.

### Modifications

**`packages/react-frontend/src/pages/GamePage.jsx`**
- Replace individual `/api/cards/next` calls with `/api/cards/all` so that all cards load in bulk
- Implement deck-based card serving that uses `deckUtils` functions
- Add deck initialization and management in `useEffect` so that cards load on game start
- Handle deck exhaustion with automatic reshuffle so that gameplay continues indefinitely
- Modify `handleGuess` to draw next card from local deck so that API calls are eliminated

**`packages/react-frontend/src/utils/deckUtils.js`**
- Add `shuffleDeck()` function that uses Fisher-Yates algorithm for proper randomization
- Add `drawCard()` function that removes and returns cards from deck end
- Ensure immutable operations so that original arrays remain unchanged

**`packages/react-frontend/tests/pages/GamePage.test.jsx`**
- Update tests to work with new deck-based initialization flow
- Mock `deckUtils` functions so that test behavior is predictable
- Add comprehensive test coverage for deck integration scenarios so that edge cases are covered
- Test deck exhaustion and reshuffle functionality so that long gameplay sessions work correctly

### Testing Considerations

**What has been tested:**
- All existing tests pass with new deck management system
- Deck initialization from `/api/cards/all` endpoint
- Sequential card serving that prevents duplicates during single session
- Proper deck exhaustion and reshuffle behavior so that games continue smoothly
- Game flow continues correctly with deck-based card management
- Error handling when card fetching fails

**What the reviewer should verify:**
- Start a new game and verify that no duplicate cards appear during extended play
- Confirm that game performance remains smooth with local deck management
- Test that deck properly reshuffles when exhausted so that gameplay continues
- Verify that game initialization still works correctly if API is slow or fails
- Check that card randomization feels natural and unpredictable
- Ensure that score tracking and game progression work identically to before

### CODE REVIEW

File: packages/react-frontend/src/pages/GamePage.jsx [hunk 1]
```diff
@@ -94,24 +107,40 @@ function GamePage() {
...
+            setDeck(newDeck);
+          } else {
+            // Reshuffle deck when exhausted
+            const allCards = await apiRequest("/api/cards/all");
+            const reshuffledDeck = shuffleDeck(allCards);
+            const newCard = reshuffledDeck[reshuffledDeck.length - 1];
+            const newDeck = reshuffledDeck.slice(0, -1);
+            setCurrentCard(newCard);
+            setDeck(newDeck);
+          }
```
[the remark this answers is not in the document]
Comment (kevinrutledge) [reply packages/react-frontend/src/pages/GamePage.jsx:131]: Overkill. This avoids single level of abstraction principles and makes it less clear.

**Comment by Ven909** [comment]
**IE5 code review**
I have no comments. The code looks all good to me. It's ready to be merged.

**Review by Ven909 — APPROVED** [review]
**IE5 code review**
I have no comments. The code looks all good to me. It's ready to be merged.
